proposals/selfAssemblingModelHierarchy/README.md

```diff
@@ -0,0 +1,342 @@
+
+### Model Hierarchy Complexity
+The complexity of model hierarchy maintenance can lead users to the following workarounds.
+* Some users flood the scene graph with `group` tags and make all `Xform` prims `group`s (erroneously including those specified under `component` models).
+* Some users build model-discovery features around direct calls to `GetKind()` instead of the cached `IsKind()` to workaround model hierarchy rules, paying for the caching without benefiting and potentially leading to inconsistencies with how `kind` is handled across the ecosystem. Users may believe tooling that relies on the unvalidated `GetKind()` behaves more correctly.
+* Finding model hierarchy unreliable or complicated, some users begin to build model-discovery features around composition arc presence. `kind` as a first class feature was designed in part to aid in discovery of important prims _without composition introspection_ and to preserve behavior across flattening of arcs.
+
+### Model Hierarchy Impacting Imaging
+Model hierarchy originally could not affect the rendered result and so the consequences of incorrect model hierarchy patterns were minimal. However, the new pattern based collection [proposal](https://github.com/PixarAnimationStudios/USD-proposals/pull/4) aims to leverage model hierarchy in its predicates. Model hierarchy will now affect (and potentially accelerate) collection membership computation. **Tagging a prim as an `assembly` or a `component` and failing to maintain proper `group` tagging could change the results of material bindings or light linking.**
+
+### Model Hierarchy and Namespace Editing
+Recent additions to OpenUSD include the namespace editor and relocates. Scene graph restructuring may become more common as a result of these utilities, and with it model hierarchy invalidations.
+
+## Proposal
+The USD Glossary provides this guidance on model hierarchy maintenance:
+
+> When assets are published/packaged with model kinds already established, the model hierarchy becomes mostly “self assembling”.
+
+As the above examples demonstrate, a self assembling model hierarchy unfortunately has been complicated by the need for the "glue" `group` kind.
+
+There is prior art for composition "fixing" errant model hierarchy. Consider just the department store. It's not uncommon to see users (incorrectly) apply the `group` kind to all `Xform` and `Scope` prims in a hierarchy.
+```
+def Xform "DepartmentStore" (kind = "component") {
+    def Scope "GroundFloor" (kind = "group") {
+        def Scope "Housewares" (kind = "subcomponent") {
+        }
+    }
+}
+```
+
+The authored `group` kind is ignored with respect to model hierarchy because its parent isn't a `group` or `assembly`. With minimal trade-offs, this proposal argues that just as incorrect usage can be discarded by composition, _correct `group` usage can be propagated by composition_.
```

Comment:
Technically, it is _population_ that is ignoring these errant `kind` tags, not composition.